.github/actions/install-deps/action.yml

```diff
@@ -0,0 +1,98 @@
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+name: Install dependencies
+description: Install system dependencies needed for OpenTitan
+
+inputs:
+  verilator-version:
+    description: Verilator version to install
+    required: true
+    default: '4.210'
+  verilator-path:
+    description: Path at which to install Veriltator
+    required: true
+    default: /tmp/verilator
+  verible-version:
+    description: Verible version to install
+    required: true
+    default: 'v0.0-3622-g07b310a3'
+  verible-path:
+    description: Path at which to install Verible
+    required: true
+    default: /tmp/verible
+  apt-archive-path:
+    description: Path at which to download APT package archives
+    required: true
+    default: /tmp/apt-archives
+
+runs:
+  using: composite
+  steps:
+    - name: Enable caching for APT packages
+      uses: actions/cache@v4
+      with:
+        # We cache the directory where APT downloads `.deb` archives for packages.
+        path: ${{ inputs.apt-archive-path }}
+        key: ${{ runner.os }}-${{ hashFiles('apt-requirements.txt') }}
+      id: cache-apt
```

Comment:
Hmm. If we run this action on self-hosted runners then it would be downloading and uploading to Azure from Google Cloud?

Reply:
Should I disable this when self-hosted?

I can add `if ${{ runner.environment == 'github-hosted' }}` to the cache step

Reply:
Actually, the apt cache doesn't seem to even give us much improvement since the downloading was not the slowest part, the installing is. I'll remove this as well.